### PR TITLE
Reduce slowness due to ES pre-checks

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

This PR removes a lot of checks following this one https://github.com/akeneo/pim-community-dev/pull/7821 which removed 
- Check category existence before using them in ES

This PR removes as well:
- Price: Check that currency exists and is active before filtering on them
- Ancestor: A ProductModel Id exists before filtering on them
- Family : Check that families exists before filtering on them
- Group: Check that groups exists before filtering on them
- Parent: Check that a ProductModel id exists before filtering on them
- SelfAndAncestorFilter: Check that a ProductModel id and a Product exist before filtering on them
- WithAttributes: Check that the attributes exists before filtering on them

We did that in order to have feedback on the API in case of querying a non-existent things.
As it is not documented we remove it to  have better performance and to be consistent with the previous PR I made.

The consequencies it is that it will return nothing in case of a wrong queryied value (only in API use case as the other use cases didn't used it)

This impacts the PQB so PO may be aware of that, Nolwenn is Ok for the API.
But import / **rules** etc may be impacted.

In CE the possible points of failure are (I tried to remove the API impacts/ datagrid impacts / and filter where we check before trying to apply a filter) :
- RemoveProductCommand in Catalog
- GroupUpdater
- FilterDataValidator.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | OK
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | OK for Nono
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/family/remove_attribute_from_a_family.feature [hunk 1]
```diff
@@ -8,29 +8,19 @@ Feature: Remove attribute from a family
     Given the "catalog_modeling" catalog configuration
     And I am logged in as "Peter"
 
-  Scenario: Successfully remove an attribute from a family and display it as removable from product
+  Scenario: Successfully remove an attribute from a family and assert its removal from product, product-model, variant-product
```
Comment (anaelChardan) [inline features/family/remove_attribute_from_a_family.feature]: It uses the UI only to delete
The check is now done through repository

**Comment by jjanvier** [comment]
I know I'll be really annoying, but the title of your PR shouldn't be "Reduce slowness due to ES pre-checks".
Why? Simply because with this title I don't know if it's something the product owners are aware of or if if it's just a the wish of a developer.

Also, please explain the consequences of this PR (are there consequences on the API?, on the imports?).

**Comment by anaelChardan** [comment]
It was explained(at least some parts) in the PR referenced in the beginning ;)

**Comment by jjanvier** [comment]
In the PR referenced in the beginning, there is no explanation of what functionalities are impacted by this change. So please, I'll ask it again, list all the functional consequences of this change. And make sure the POs are aware of those changes. I really doubt only the API is impacted.

**Pull request closed without merging**